lib/Parse/ParseExpr.cpp

```diff
@@ -1869,12 +1869,18 @@ DeclName Parser::parseUnqualifiedDeclName(bool afterDot,
 
 static bool shouldAddSelfFixit(DeclContext* Current, DeclName Name,
                                DescriptiveDeclKind &Kind) {
-  if (Current->isTypeContext() || !Current->getInnermostTypeContext())
+  auto InnermostTypeContext = Current->getInnermostTypeContext();
+  if (Current->isTypeContext() || !InnermostTypeContext)
     return false;
+
   if (auto *Nominal = Current->getInnermostTypeContext()->
       getAsNominalTypeOrNominalTypeExtensionContext()){
     // FIXME: we cannot resolve members appear later in the body of the nominal.
-    auto LookupResults = Nominal->lookupDirect(Name);
+    
+    // Fixes a crasher where a lookup for a type in a local context would
+    // crash inside ASTMangler
+    auto IgnoreExtensions = InnermostTypeContext->getLocalContext() != nullptr;
+    auto LookupResults = Nominal->lookupDirect(Name, IgnoreExtensions);
```

Comment:
Probably further down, to whatever decides whether we need to go look for loaded extensions, but yes. (Don't forget a test case with a type inside a type inside a function.)